### DISCUSSION TITLE
Wt32-eth

### DISCUSSION THREAD

**Discussion by Francesco-Rugg** [forkineye/ESPixelStick#605] [Q&A]
Can I test [ESPixelStick v4.0-beta4](https://github.com/forkineye/ESPixelStick/releases/tag/v4.0-beta4)
For WT32-ETH01?

**Comment by MartinMueller2003** [comment]
that is a supported platform. I found and fixed a crash that is present in b4 so you may want to try the latest build  on the google drivr

**Reply by Francesco-Rugg** [reply]
Ok, thank you. Where I can find the google drive?

**Comment by MartinMueller2003** [comment]
Here is the latest build.
dist.zip


| 
| 
|  | 
dist.zip


 |

 |

 |




    On Tuesday, December 13, 2022 at 04:56:13 AM EST, Francesco-Rugg ***@***.***> wrote:  
 
 


Ok, thank you. Where I can find the google drive?

**Reply by Francesco-Rugg** [reply]
@MartinMueller2003  so is the same dist.zip tant you already sent to me?

**Reply by Francesco-Rugg** [reply]
Sorry, but I'm a newbie and I don't know how to use github. 
I can't download any dust.zip from your answer

**Comment by forkineye** [comment]
Any reason not to use the CI builds instead? https://github.com/forkineye/ESPixelStick/actions/workflows/build.yaml

**Comment by MartinMueller2003** [comment]
Hi Shelby

Three reasons:

1. The CI builds are just images, not integrated into a dist.zip file and I am seeing a lot of people that have no idea how to use an image file or integrate it into a dist.zip.
2. In this case, the CI build and my latest changes are in sync. Very often, my builds are ahead of the CI builds (this is NOT an issue that should be addressed, It is good to have the check step).
3. The binary files do not include the FS file set so they cannot be used on their own.

So I have gotten into the habit of making sure the dist.zip on the google drive is always up-to-date and then point people to it. If the CI build also generated a dist.zip then things would get simpler and we could simply point people to it. It would be really great if the CI build ran on any push to a derived branch and created a dist on that branch. I have a script that comes close. It puts all generated bin files into a firmware dir as part of the build process but I still manually move it to the dist dir to make the zip file

**Reply by forkineye** [reply]
The Release Archive has everything including the flash tool with versioning. It's the same artifact that's used for releases. The CI builds are generated on every PR as well.

**Comment by MartinMueller2003** [comment]
It looks like the link to the google drive was damaged.
Here it is again.
https://drive.google.com/file/d/1DG3wz8lbYNc371OIbeOqkzMaY5Zd2SKA/view?usp=sharing

**Reply by Francesco-Rugg** [reply]
Ok if I have any issue where I have to post it? In this repository or where?

**Comment by MartinMueller2003** [comment]
Download the file from google drive. unzip it in any location you like. Read the readme in the unziped directory. Connect your device to the PC using a USB cable, start the flashtool. fill in the fields and upload the image

**Comment by MartinMueller2003** [comment]
Yes you can post your questions in this forum.

**Comment by Francesco-Rugg** [comment]
Thank you! @MartinMueller2003

**Comment by Francesco-Rugg** [comment]
This came out when try to flash wt32
![wt32](https://user-images.githubusercontent.com/62062196/207338609-87223104-8c47-48da-b76f-2c1afc825eca.JPG)
I use the same connection that I used for flashing wled to this board so I don't think thats a connection issue

**Reply by MartinMueller2003** [reply]
The line with "Connecting  ....____....____ is telling you that the device is not in programming mode. You needed to reset the device with GPIO 0 (zero) held low. You will need to do that three times.

**Reply by Francesco-Rugg** [reply]
So I need to erase flash three times before start flashing new ones?

**Comment by Francesco-Rugg** [comment]
I use online esptool for erasing and in it I can erase the flash 
[https://espressif.github.io/esptool-js/](url)

**Comment by MartinMueller2003** [comment]
My WT32 does not have a USB interface and I have to manually put the device into programming mode three times when using the FlashTool.

**Reply by Francesco-Rugg** [reply]
I also use a programmer and my wt32 doesn't have USB connection

**Reply by Francesco-Rugg** [reply]
> My WT32 does not have a USB interface and I have to manually put the device into programming mode three times when using the FlashTool.

What do you mean when you say "three times"

**Comment by MartinMueller2003** [comment]
The flash tool performs three operations: Clear flash, write fs, write program. If the USB is present then the device is reset into programming mode three times. If there is no USB on the device then you need to pull gpio0 low (and keep it that way). Then reset the device each time you see the "connecting ..." message coming out. After it is done you remove the wire on GPIO 0 and do a final reset

**Reply by Francesco-Rugg** [reply]
I use a programmer CH340 USB TO TTL, si I need to erase the wt32 flash three times before start flashing new software is it right? 
The connections are:
+5 to +5
GND to GND
Rx to Tx
Tx to Rx
GND (on board) to IO1

**Comment by MartinMueller2003** [comment]
No. You do not need to explicitly erase the flash. You need to put the device into programming mode every time you see the "Connecting ....." message. When using the flash tool you will see that message three times and you should do a reset (with gpio0 low) if you see the dots and dashes coming out.

**Comment by Francesco-Rugg** [comment]
@MartinMueller2003 Sorry but I receiving the same error, while in wled web istaller I don't have this issue.

I see that the erase function works but the firmware flash don't go ahead
![w](https://user-images.githubusercontent.com/62062196/207430879-9c34072b-445a-4b90-9db9-36caf49543ec.JPG)

**Comment by MartinMueller2003** [comment]
That means the first step passed and you failed to put the device back into programming mode for the next step. I just did it using my ET32  and it is a pita. I had to power cycle the device three times in the process and you have to do it at the right time and quickly.

**Comment by MartinMueller2003** [comment]
You can now flash using https://espixelstickwebflasher.from-ct.com:5000/
